Add bitbox02

reardencode opened the pull request:
###### Motivation for this change

Preparatory work for supporting BitBox02 in electrum whenever electrum 4.0 comes out.

Removes python2 support from `block-io`, `streamlit`, and `zeronet`, hopefully we're not caring about losing python2 support these days?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jonringer commented on maintainers/maintainer-list.nix:
```diff
@@ -6428,6 +6428,12 @@
     githubId = 1973389;
     name = "Reuben D'Netto";
   };
+  reardencode = {
+    email = "freedom@reardencode.com";
+    github = "reardencode";
+    githubId = 730881;
+    name = "Brandon Black";
+  };
```
Please make this into a separate commit:

````
maintainers: add <name>
```

reardencode replied:
done

jonringer commented on pkgs/top-level/python-packages.nix:
```diff
@@ -6821,6 +6821,10 @@ in {
 
   ckcc-protocol = callPackage ../development/python-modules/ckcc-protocol { };
 
+  bitbox02 = callPackage ../development/python-modules/bitbox02 { };
+
+  noiseprotocol = callPackage ../development/python-modules/noiseprotocol { };
+
```
please attempt to sort these with the other packages, don't want to contribute more to the mess

jonringer reviewed (commented):
each package change should be it's own commit, I see 2 package additions, 1 bump, 1 modification, and one maintainer addition. so there should roughly be 5 commits

To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

nyanloutre commented:
@ofborg build python3Packages.block-io python3Packages.py-multihash zeronet streamlit

reardencode committed:
py-multihash: Accept base58 >=2.0 as dependency

It looks like py-multihash depends on base58 < 2.0 because it wants to
support python3.4, but base58 2.0 supports python >= 3.5 only. This
doesn't effect nixpkgs because we don't have python3.4.

reardencode committed:
base58: Version bump

reardencode committed:
bitbox02: new python package

reardencode committed:
maintainers: add me (reardencode)

reardencode committed:
noiseprotocol: new python package

reardencode commented:
@nyanloutre I'm quite new to contributing to nixpkgs, is there a proper procedure for dropping python2 support from `streamlit`, `zeronet`, and `block-io`?  I don't think anything changed for us with `py-multihash`

Thanks!

prusnak commented on pkgs/development/python-modules/bitbox02/default.nix:
```diff
@@ -0,0 +1,35 @@
+    sha256 = "1mcbg7bvlsii1kj2mm5686z01pwcc36qbqlhz4a8syjrzjm6pl2k";
+  };
+
+  doCheck = false;
```
Please add a comment why you skip tests or fix the tests, so they don't need to be skipped.

prusnak commented on pkgs/development/python-modules/bitbox02/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  meta = with stdenv.lib; {
+    description = "Communicate with your BitBox02 using Python";
+    homepage = https://github.com/digitalbitbox/bitbox02-firmware;
```
Use quotes for URLs (as per RFC45)

prusnak commented on pkgs/development/python-modules/bitbox02/default.nix:
```diff
@@ -0,0 +1,35 @@
+, semver
+, typing-extensions
+, base58
+, lib
```
Drop `lib`, it is not used anywhere.

prusnak commented on pkgs/development/python-modules/noiseprotocol/default.nix:
```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, buildPythonPackage
+, lib
```
Drop `lib`, it's not used anywhere.

prusnak commented on pkgs/development/python-modules/noiseprotocol/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  meta = with stdenv.lib; {
+    description = "Python Implementation of Noise Protocol Framework";
+    homepage = https://github.com/plizonczyk/noiseprotocol;
```
Use quotes for URLs (as per RFC45)

prusnak commented on pkgs/development/python-modules/bitbox02/default.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Communicate with your BitBox02 using Python";
+    homepage = https://github.com/digitalbitbox/bitbox02-firmware;
+    license = licenses.asl20;
+    maintainers = [ maintainers.reardencode ];
```
```suggestion
    maintainers = with maintainers; [ reardencode ];
```

prusnak commented on pkgs/development/python-modules/noiseprotocol/default.nix:
```diff
@@ -0,0 +1,37 @@
+    description = "Python Implementation of Noise Protocol Framework";
+    homepage = https://github.com/plizonczyk/noiseprotocol;
+    license = licenses.mit;
+    maintainers = [ maintainers.reardencode ];
```
```suggestion
    maintainers = with maintainers; [ reardencode ];
```

prusnak commented:
@reardencode Can you please rebase the PR, there are conflicts. While you are at it, please check whether some of the packages cannot be updated to even more recent versions. Finally, please prefix your commit messages with `python3Packages.`  so it reads `python3Packages.base58` instead of `base58`. Thanks!

prusnak reviewed (changes requested):
@reardencode Can you please rebase the PR, there are conflicts. While you are at it, please check whether some of the packages cannot be updated to even more recent versions. Finally, please prefix your commit messages with `python3Packages.`  so it reads `python3Packages.base58` instead of `base58`. Thanks!
